edu-services/cm-service/cm-api/api/src/java/org/sakaiproject/coursemanagement/api/CourseSitePublishService.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2015 Apereo Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *             http://opensource.org/licenses/ecl2
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sakaiproject.coursemanagement.api;
+
+
+/**
+ * sakai service for publishing course sites a set number of days before a term begins.
+ * <p>
+ * example: <br/>
+ * <code>
+ *     fall term, 2008: starts August 20th, 2008
+ *
+ *     let's say that some course sites, including bio 201, are created for the fall term 2008.
+ *     assume that we want the course sites to be published two weeks before the fall term starts.
+ *     thus, we want the sakai site for bio 201 to be published on August 20th - 14 days = August 6th, 2008.
+ * </code>
+ * </p>
+ */
+public interface CourseSitePublishService {
+
+   // permissions
+   public final static String PERMISSION_COURSE_SITE_PUBLISH     = "course_site_publish_service.publish";
+
+   // site property
+   public final static String SITE_PROPRTY_COURSE_SITE_PUBLISHED = "course_site_publish_service.publish.set";
```

Comment:
Nitpicking, but I'd prefer 'PROPERTY' spelled out. IMHO it's easier and less syntax error prone to type out familiar words than to skip particular letters.